### PR TITLE
remove unwanted control-flow code from DeBERTa-V2

### PR DESCRIPTION
**Pull request by kamalkraj**
# What does this PR do?

Removes never executed branch from `deberta-v2` code

discussed in https://github.com/huggingface/transformers/pull/13120#issuecomment-899865394

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@LysandreJik @patrickvonplaten @Rocketknight1

### CODE REVIEW

File: src/transformers/models/deberta_v2/modeling_deberta_v2.py [hunk 1]
```diff
@@ -753,8 +753,6 @@ def disentangled_attention_bias(self, query_layer, key_layer, relative_pos, rel_
                 r_pos = relative_pos
 
             p2c_pos = torch.clamp(-r_pos + att_span, 0, att_span * 2 - 1)
-            if query_layer.size(-2) != key_layer.size(-2):
```
Comment (patrickvonplaten) [inline src/transformers/models/deberta_v2/modeling_deberta_v2.py:756]: All 

      ``` self.query_proj = nn.Linear(config.hidden_size, self.all_head_size, bias=True)
        self.key_proj = nn.Linear(config.hidden_size, self.all_head_size, bias=True)
        self.value_proj = nn.Linear(config.hidden_size, self.all_head_size, bias=True)```

have the same dim which is why this control flow is never true no? 

LGTM to remove for me

Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_deberta_v2.py:756]: > have the same dim which is why this control flow is never true no?

Yes

Comment (kamalkraj) [reply src/transformers/models/deberta_v2/modeling_deberta_v2.py:756]: `Deberta-v2` same dim always. so the control flow is never true

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @kamalkraj 

cc @BigBird01
